airbyte-integrations/connectors/source-hubspot/source_hubspot/api.py

```diff
@@ -0,0 +1,394 @@
+                if "next" in response["paging"]:
+                    params["after"] = response["paging"]["next"]["after"]
+                else:
+                    break
+            else:
+                if not response.get(self.more_key, False):
+                    break
+                if self.page_field in response:
+                    params[self.page_filter] = response[self.page_field]
+
+    def read_chunked(self, getter: Callable, params: Mapping[str, Any] = None):
+        params = {**params} if params else {}
+        now_ts = int(pendulum.now().timestamp() * 1000)
+        start_ts = int(self._start_date.timestamp() * 1000)
+
+        for ts in range(start_ts, now_ts, self.chunk_size):
+            end_ts = ts + self.chunk_size
+            params["startTimestamp"] = ts
+            params["endTimestamp"] = end_ts
+            logger.info(f"Reading chunk from {ts} to {end_ts}")
+            yield from self.read(getter, params)
+
+    @property
+    def properties(self) -> Mapping[str, Any]:
+        if not self.entity:
+            return {}
+
+        props = {}
+        data = self._api.get(f"/properties/v2/{self.entity}/properties")
+        for row in data:
+            props[row["name"]] = {"type": row["type"]}
+
+        return props
+
+
+class CRMObjectsAPI(StreamAPI, ABC):
+    data_path = "results"
+
+    @property
+    @abstractmethod
+    def url(self):
+        """Endpoint URL"""
+
+    def __init__(self, include_archived_only=False, **kwargs):
+        super().__init__(**kwargs)
+        self._include_archived_only = include_archived_only
+
+    def list(self, fields) -> Iterable:
+        params = {
+            "archived": str(self._include_archived_only).lower(),
+        }
+        for record in self.read(partial(self._api.get, url=self.url), params):
+            yield record
+
+
+class CampaignsAPI(StreamAPI):
+    entity = "campaign"
+    more_key = "hasMore"
+    data_path = "campaigns"
+    limit = 500
+
+    def list(self, fields) -> Iterable:
+        url = "/email/public/v1/campaigns/by-id"
+        for row in self.read(getter=partial(self._api.get, url=url)):
+            record = self._api.get(f"/email/public/v1/campaigns/{row['id']}")
+            yield record
+
+
+class CompaniesAPI(CRMObjectsAPI):
+    entity = "company"
+    url = "/crm/v3/objects/companies"
+
+
+class ContactListsAPI(CRMObjectsAPI):
+    url = "/crm/v3/objects/contacts"
+
+
+class ContactsAPI(CRMObjectsAPI):
+    entity = "contact"
+    url = "/crm/v3/objects/contacts"
+
+
+class DealsAPI(CRMObjectsAPI):
+    entity = "deal"
+    url = "/crm/v3/objects/deals"
+
+
+class LineItemsAPI(CRMObjectsAPI):
+    entity = "line_item"
+    url = "/crm/v3/objects/line_items"
+
+
+class ProductsAPI(CRMObjectsAPI):
+    entity = "product"
+    url = "/crm/v3/objects/products"
+
+
+class QuotesAPI(CRMObjectsAPI):
+    entity = "quotes"
+    url = "/crm/v3/objects/quotes"
+
+
+class TicketsAPI(CRMObjectsAPI):
```

Comment:
why does every class have `API` in the name? Shouldn't it just be the `API` class and then everything else is a stream?